Реализовано ТЗ 15 спринта.

Elena-Zhorzhevna opened the pull request:
Добавлена работа с базой данных, добавлена функция бронирования вещей пользователями.

Elena-Zhorzhevna committed:
Реализовано ТЗ 15 спринта.
Добавлена работа с базой данных, добавлена функция бронирования вещей пользователями.

Elena-Zhorzhevna committed:
Добавлен файл с настройками бд h2 для тестового класса.

Elena-Zhorzhevna committed:
Изменен путь к файлу с настройками тестового класса.

kriwsha commented on src/main/java/ru/practicum/shareit/booking/mapper/BookingMapper.java:
```diff
@@ -0,0 +1,61 @@
+@Component
+public class BookingMapper {
+    private final UserServiceImpl userService;
+    private final ItemServiceImpl itemService;
```
1. Не забывайте про полиморфизм - мы должны использовать интерфейсы, а не реализации
2. Ни UserService, ни ItemService не используется в этом классе - их следует удалить отсюда

kriwsha commented on src/main/java/ru/practicum/shareit/booking/mapper/BookingMapper.java:
```diff
@@ -0,0 +1,61 @@
+/**
+ * Класс для преобразования объектов типа Booking в тип BookingDto и обратно.
+ */
+@Component
```
У вас утилитарный класс. Доступ идет не через ссылку, а через имя класса. Поэтому не следует его делать компонентом

kriwsha commented on src/main/java/ru/practicum/shareit/booking/mapper/BookingMapper.java:
```diff
@@ -0,0 +1,61 @@
+        return dto;
+    }
+
+    public Booking mapBookingDtoToPutToBooking(BookingDtoToPut bookingDtoToPut, Long bookerId) {
```
Оба метода можно сделать статическими, чтобы не смешивать в одном классе схожую логику с разным способом доступа к ней. В местах, где вы вызываете эти методы через ссылку следует заменить на статических доступ через класс. Так код будет проще и строже

kriwsha commented on src/main/java/ru/practicum/shareit/booking/model/Status.java:
```diff
@@ -0,0 +1,16 @@
+/**
+ * Статус бронирования вещи.
+ * Может иметь значения:
+ * WAITING - новое бронирование, ожидает одобрения,
```
Вообще, каждую энамку можно прокомментировать отдельно, не вынося это в общий комментарий

kriwsha commented on src/main/java/ru/practicum/shareit/item/mapper/CommentMapper.java:
```diff
@@ -0,0 +1,24 @@
+import ru.practicum.shareit.item.model.Comment;
+
+
+public class CommentMapper {
```
Здесь все правильно. Утилитарный класс с двумя статическими методами

kriwsha commented on src/main/java/ru/practicum/shareit/item/mapper/ItemMapper.java:
```diff
@@ -1,28 +1,32 @@
+                .build();
+    }
+
+    public ItemDto mapToItemDtoWithComments(Item item) {
```
А здесь один метод статический, а другой нестатический, при этом один делает обратное действие другого метода

kriwsha commented on src/main/java/ru/practicum/shareit/item/service/ItemServiceImpl.java:
```diff
@@ -38,35 +62,40 @@ public ItemServiceImpl(InMemoryItemStorage itemStorage, UserService userService)
         log.info("Запрос на получение всех вещей.");
-        return itemStorage.getAll().stream().map(ItemMapper::mapToItemDto).toList();
+        return itemRepository.findAll().stream()
+                .map(itemMapper::mapToItemDtoWithComments)
```
Вот здесь можно написать ItemMapper::mapToItemDtoWithComments, и будет то же самое, поэтому создание экземпляра класса необязательно

Elena-Zhorzhevna committed:
Вспомнила про полиморфизм.
Удалила неиспользуемые сервисные классы из BookingMapper.
Убрала ненужные аннотации, чтобы не делать утилитарные классы компонентом.
Сделала все методы мапперных классов статическими.
Каждую энамку прокомментировала отдельно.
